### ISSUE TITLE
Crash at exit (Regression App SDK 1.3)

### ISSUE THREAD

**Issue by jschwizer99** [microsoft/microsoft-ui-xaml#8409]
### Describe the bug

When fail-fast is enabled a Close() triggered via Menuflyout command will trigger a crash for App SDK 1.3.

```
Exception thrown at 0x00007FFA125ECB69 (KernelBase.dll) in TreeStability.exe: WinRT originate error - 0x80000013 :
  'The given object has already been closed / disposed and may no longer be used.'.
The thread 0x6bfc has exited with code 0 (0x0).
Microsoft.UI.Xaml.dll!00007FF930CF304E: CallContext:[\ImageDecodeActivity] 8001010E - RPC_E_WRONG_THREAD
Unhandled exception at 0x00007FFA126D7CD2 (KernelBase.dll) in TreeStability.exe: 0xC000027B:
  An application-internal exception has occurred (parameters: 0x00000220040D6540, 0x0000000000000001).
```

When fail-fast is disabled, suspicious exceptions are visible, but the application closes with exit code 0.
```
//#if _DEBUG
//    if (IsDebuggerPresent())
//    {
//        this->DebugSettings().FailFastOnErrors(true);
//    }
//#endif
```

```
Exception thrown at 0x00007FFA125ECB69 (KernelBase.dll) in TreeStability.exe: WinRT originate error - 0x80000013 : 
  'The given object has already been closed / disposed and may no longer be used.'.
The thread 0x5cec has exited with code 0 (0x0).
Microsoft.UI.Xaml.dll!00007FF929DC304E: CallContext:[\ImageDecodeActivity] 8001010E - RPC_E_WRONG_THREAD
Exception thrown at 0x00007FFA125ECB69 in TreeStability.exe: Microsoft C++ exception: winrt::hresult_wrong_thread
  at memory location 0x00000005B02FDFC0.
The thread 0x1654 has exited with code 0 (0x0).
The thread 0x4ef8 has exited with code 0 (0x0).
The thread 0x317c has exited with code 0 (0x0).
The thread 0x55a8 has exited with code 0 (0x0).
The thread 0x7654 has exited with code 0 (0x0).
Exception thrown at 0x00007FFA125ECB69 (KernelBase.dll) in TreeStability.exe: WinRT originate error - 0x80000013 :
   'The given object has already been closed / disposed and may no longer be used.'.
onecoreuap\windows\moderncore\inputv2\inputhost\components\dragdrop\lib\closeableobject.cpp(31)\
  Microsoft.UI.Input.dll!00007FF9335CF1A1: (caller: 00007FF9335CF022) ReturnHr(1) tid(6580) 80000013
  The object has been closed.
The thread 0x5e90 has exited with code 0 (0x0).
The thread 0x13e8 has exited with code 0 (0x0).
The thread 0x4a88 has exited with code 0 (0x0).
The thread 0x6224 has exited with code 0 (0x0).
The thread 0x6f98 has exited with code 0 (0x0).
The thread 0x4324 has exited with code 0 (0x0).
The thread 0x790 has exited with code 0 (0x0).
The thread 0x69b0 has exited with code 0 (0x0).
The thread 0x56a0 has exited with code 0 (0x0).
The thread 0x5a7c has exited with code 0 (0x0).
The thread 0x70a0 has exited with code 0 (0x0).
The thread 0x630 has exited with code 0 (0x0).
The thread 0x389c has exited with code 0 (0x0).
The program '[7412] TreeStability.exe' has exited with code 0 (0x0).

```

The observed failure pattern looks like the one reported in https://github.com/microsoft/microsoft-ui-xaml/issues/8395. However, in this case, no webview2 is involved.
We have a large application that unfortunately also shows the crash with fail-fast disabled.  This blocks us from upgrading from 1.25 to 1.3.
Closing the window will work fine for the small example app. However, on the large application, we also observe the crash in this condition.
@Scottj1s / @DmitriyKomin I've mentioned this observation during the last April Community Call. As it does not fully match the reported #8395 I wrote a new ticket. It could be that both observations could be two manifestations of a bigger 1.3 regression.

### Steps to reproduce the bug

- Unpack [TreeStability.zip](https://github.com/microsoft/microsoft-ui-xaml/files/11277025/TreeStability.zip)
- Compile and start in Debug x64
- If you get a deployment error close and reopen VS (Is caused by stripping away the packages folder)
- Close application with Menuflyout button -> crash
- Close application with Button -> no crash

Comment out fast-fail -> no crash

```
 	KernelBase.dll!RaiseFailFastException()	Unknown
 	combase.dll!RoFailFastWithErrorContextInternal2(HRESULT hrError, unsigned long cStowedExceptions, _STOWED_EXCEPTION_INFORMATION_V2 * * aStowedExceptionPointers) Line 1455	C++
 	Microsoft.ui.xaml.dll!OnFailureEncountered()	Unknown
 	Microsoft.ui.xaml.dll!OnFailure<866>()	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::CompositionTarget::GetCompositorForCurrentThread()	Unknown
 	Microsoft.UI.Xaml.Controls.dll!winrt::impl::call_factory_cast<winrt::Microsoft::UI::Composition::Compositor (__cdecl*)(winrt::Microsoft::UI::Xaml::Media::ICompositionTargetStatics const &),winrt::Microsoft::UI::Xaml::Media::CompositionTarget,winrt::Microsoft::UI::Xaml::Media::ICompositionTargetStatics,`winrt::Microsoft::UI::Xaml::Media::CompositionTarget::GetCompositorForCurrentThread'::`2'::<lambda_1>>()	Unknown
 	Microsoft.UI.Xaml.Controls.dll!LifetimeHandler::TryGetMaterialHelperInstance(void)	Unknown
 	Microsoft.UI.Xaml.Controls.dll!MaterialHelper::NoiseChanged(struct winrt::event_token)	Unknown
 	Microsoft.UI.Xaml.Controls.dll!AcrylicBrush::~AcrylicBrush(void)	Unknown
 	Microsoft.UI.Xaml.Controls.dll!winrt::impl::heap_implements<class AcrylicBrush>::`vector deleting destructor'(unsigned int)	Unknown
 	Microsoft.UI.Xaml.Controls.dll!ReferenceTracker<class AcrylicBrush,winrt::Microsoft::UI::Xaml::Media::implementation::AcrylicBrushT>::DeleteInstanceOnUIThread(class std::unique_ptr<class AcrylicBrush,struct std::default_delete<class AcrylicBrush> > &&)	Unknown
 	Microsoft.UI.Xaml.Controls.dll!winrt::implements<class AcrylicBrush,struct winrt::Microsoft::UI::Xaml::Media::AcrylicBrush,struct winrt::Microsoft::UI::Xaml::Media::IAcrylicBrush2,struct winrt::composable,struct winrt::composing,struct winrt::Microsoft::UI::Xaml::Media::IXamlCompositionBrushBaseOverrides,struct winrt::Microsoft::UI::Xaml::Media::IBrushOverrides,struct IReferenceTrackerExtension>::Release(void)	Unknown
 	Microsoft.ui.xaml.dll!ctl::release_interface<class DirectUI::DependencyObject>(class DirectUI::DependencyObject * &)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DependencyObject::ReleaseForPeerReferenceHelper()	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DependencyObject::ClearPeerReferences()	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DependencyObject::DisconnectFrameworkPeerCore()	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DependencyObject::DisconnectFrameworkPeer(bool)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DependencyObject::EndShutdown(void)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DXamlCore::ShutdownAllPeers(void)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DXamlCore::CommonShutdown(void)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DXamlCore::DeinitializeInstance(void)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::DXamlCore::Deinitialize(enum DeinitializationType)	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::WindowsXamlManager::XamlCore::Close(void)	Unknown
 	Microsoft.ui.xaml.dll!Microsoft::WRL::Details::DelegateArgTraits<long (__cdecl ABI::Microsoft::UI::Dispatching::IDispatcherQueueHandler::*)(void)>::DelegateInvokeHelper<Microsoft::WRL::Implements<Microsoft::WRL::RuntimeClassFlags<2>,ABI::Microsoft::UI::Dispatching::IDispatcherQueueHandler,Microsoft::WRL::FtmBase>,`DirectUI::WindowsXamlManager::EnqueueClose'::`2'::<lambda_1> &,1>::Invoke()	Unknown
 	CoreMessagingXP.dll!Microsoft::UI::Dispatching::DispatcherQueue::DeferInvokeCallback(void *)	Unknown
 	CoreMessagingXP.dll!CFlat::SehSafe::Execute<<lambda_a81ff790741c2a62f2197c2561f5fe49>>()	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::ActionCallback::ImportAdapter$(class CFlat::Box$1<struct CFlat::FunctionPointerAndUserData$1<long > > *)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Messaging::MessageSession::Callback_InvokeDeferInvoke(class Microsoft::CoreUI::ActionCallback *,struct System::UIntPtr)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::DeferredCall::Callback_Dispatch(void)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::DeferredCallDispatcher::Callback_OnDispatch(void)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::Dispatcher::Callback_DispatchNextItem(class Microsoft::CoreUI::Dispatch::DispatchItem *)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::Dispatcher::Callback_DispatchLoop(enum Microsoft::CoreUI::Dispatch::RunnablePriorityMask)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::EventLoop::Callback_RunCoreLoop(enum Microsoft::CoreUI::Dispatch::RunMode)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::UserAdapter::DrainCoreMessagingQueue(enum Microsoft::CoreUI::Dispatch::UserAdapter$UserPriority,void * *)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::UserAdapter::OnUserDispatch(bool,enum Microsoft::CoreUI::Dispatch::UserAdapter$UserPriority,void * *)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::UserAdapter::OnUserDispatchRaw(enum Microsoft::CoreUI::Dispatch::UserAdapter$UserPriority,bool,void * *)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::UserAdapter::DoWork(struct HWND__ *,enum Microsoft::CoreUI::Dispatch::UserAdapter$UserPriority,bool)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::UserAdapter::HandleDispatchNotifyMessage(struct HWND__ *,unsigned __int64,__int64)	Unknown
 	CoreMessagingXP.dll!Microsoft::CoreUI::Dispatch::UserAdapter::WindowProc(struct HWND__ *,unsigned int,unsigned __int64,__int64)	Unknown
 	user32.dll!UserCallWinProcCheckWow()	Unknown
 	user32.dll!DispatchMessageWorker()	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::FrameworkApplication::StartDesktop()	Unknown
 	Microsoft.ui.xaml.dll!DirectUI::FrameworkApplicationFactory::Start()	Unknown
>	TreeStability.exe!winrt::impl::consume_Microsoft_UI_Xaml_IApplicationStatics<winrt::Microsoft::UI::Xaml::IApplicationStatics>::Start(const winrt::Microsoft::UI::Xaml::ApplicationInitializationCallback & callback) Line 158	C++
 	TreeStability.exe!winrt::Microsoft::UI::Xaml::Application::Start::__l2::<lambda>(const winrt::Microsoft::UI::Xaml::IApplicationStatics & f) Line 12270	C++
 	TreeStability.exe!winrt::impl::factory_cache_entry<winrt::Microsoft::UI::Xaml::Application,winrt::Microsoft::UI::Xaml::IApplicationStatics>::call<void <lambda>(const winrt::Microsoft::UI::Xaml::IApplicationStatics &) &>(winrt::Microsoft::UI::Xaml::Application::Start::__l2::void <lambda>(const winrt::Microsoft::UI::Xaml::IApplicationStatics &) & callback) Line 6457	C++
 	TreeStability.exe!winrt::impl::call_factory<winrt::Microsoft::UI::Xaml::Application,winrt::Microsoft::UI::Xaml::IApplicationStatics,void <lambda>(const winrt::Microsoft::UI::Xaml::IApplicationStatics &)>(winrt::Microsoft::UI::Xaml::Application::Start::__l2::void <lambda>(const winrt::Microsoft::UI::Xaml::IApplicationStatics &) && callback) Line 6480	C++
 	TreeStability.exe!winrt::Microsoft::UI::Xaml::Application::Start(const winrt::Microsoft::UI::Xaml::ApplicationInitializationCallback & callback) Line 12271	C++
 	TreeStability.exe!wWinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, wchar_t * __formal, int __formal) Line 69	C++
 	TreeStability.exe!invoke_main() Line 123	C++
 	TreeStability.exe!__scrt_common_main_seh() Line 288	C++
 	TreeStability.exe!__scrt_common_main() Line 331	C++
 	TreeStability.exe!wWinMainCRTStartup(void * __formal) Line 17	C++
 	kernel32.dll!BaseThreadInitThunk()	Unknown
 	ntdll.dll!RtlUserThreadStart()	Unknown
```

### Expected behavior

No crash (as in App SDK < 1.3 versions)

### Screenshots

![image](https://user-images.githubusercontent.com/7672712/233187885-3d65a88f-baeb-4326-bf30-20af8c9ab710.png)

### NuGet package version

WinUI 3 - Windows App SDK Preview 1: 1.3.230228005-preview1

### Windows version

Windows 10 (21H2): Build 19044, Windows 10 (21H1): Build 19043

### Additional context

Observed and tested with App SDK 1.3 (1.3.230331000)

**Comment by codendone** [comment]
Note: The noisy exception here was fixed in That exception was fixed in 1.3.230602002. But while the noisy C++/WinRT exception no longer is thrown with that fix, FailFastOnErrors=true still observes the underlying error and crashes the app. So another fix will be needed to make FailFastOnErrors not crash.

**Comment by JesseCol** [comment]
Testing the repro app with TreeStability.zip with WinAppSDK 1.4.231115000, I'm seeing no crashes when I close the app.  (Even when I set FailFastOnErrors=true).  Closing as completed, please holler if you still see an issue.  Thanks!
